astropy/io/fits/tests/test_core.py

```diff
@@ -674,8 +674,8 @@ def test_fits_file_handle_mode_combo(self):
                 pass
 
         # This should not work since the modes conflict
-        with pytest.raises(ValueError):
-            with open(self.data("test0.fits"), "rb") as handle:
+        with open(self.data("test0.fits"), "rb") as handle:
+            with pytest.raises(ValueError):
                 with fits.open(handle, mode="ostream") as _:
```

Comment:
It might be better to use a single `with`-statement for all these context managers.